File: Retos/Reto #26 - TESTING [Media]/ruby/test0n3.rb
# Hay dos formas de probar esta función:
# - sin usar alguna librería/framework para las pruebas
# - usando alguna librería/framework para las pruebas
#
# lo relevante es tener claro los inputs y las respuestas esperadas.
# en este caso se prueban las siguientes datos:
# - fecha pasada sin viernes 13
# - fecha futura sin viernes 13
# - fecha pasada con viernes 13
# - fecha futura con viernes 13

# frozen_string_literal: true

require_relative '../../Reto #12 - VIERNES 13 [Fácil]/ruby/test0n3'
# importando código con la función para probar.

# ----------------- Sin usar librerías/framework -----------------

SOME_TESTS = { 'input' => [[2022, 4], [2024, 8], [2022, 5], [2024, 9]],
               'output' => [false, false, true, true] }.freeze

errors = 0

SOME_TESTS['input'].each_with_index do |test, index|
  resp = friday13?(test)
  expected = SOME_TESTS['output'][index]
  next if resp == expected

  errors += 1
  puts "\n\noriginal: #{test}"
  puts resp
  puts "expected: #{expected}"
end

puts "\nTests#{errors != 0 ? ' not ' : ' '}passed, #{errors} errors\n"

# ----------------- Usando librerías/framework -----------------

# gem 'minitest', '~> 5.4'
# require 'minitest/autorun'
# require 'minitest/pride'

# # class to test friday13
# class Fridat13Test < Minitest::Test
#   def test_past_not_friday13
#     # skip
#     input = [2022, 4]
#     expected = false
#     assert_equal expected, friday13?(input)
#   end

#   def test_future_not_friday13
#     # skip
#     input = [2024, 8]
#     expected = false
#     assert_equal expected, friday13?(input)
#   end

#   def test_past_friday13
#     # skip
#     input = [2022, 5]
#     expected = true
#     assert_equal expected, friday13?(input)
#   end

#   def test_future_friday13
#     # skip
#     input = [2024, 9]
#     expected = true
#     assert_equal expected, friday13?(input)
#   end
# end

File: Retos/Reto #26 - TESTING [Media]/python/test0n3.py
# código de la función le pertenece a @jcdm60
# disponible en: https://github.com/mouredev/retos-programacion-2023/blob/main/Retos/Reto%20%2312%20-%20VIERNES%2013%20%5BF%C3%A1cil%5D/python/jcdm60.py

import datetime


def is_friday_13(month, year):

    # Creamos un objeto datetime para el 13 del month y año indicados
    date = datetime.date(year, month, 13)

    # Comprobamos si es viernes (el número 4 representa el día viernes)
    if date.weekday() == 4:
        return True
    else:
        return False


def test_is_friday_13():
    tests = {'input': [[4, 2022], [8, 2024], [5, 2022], [
        9, 2024]], 'output': [False, False, True, True]}
    errors = 0

    for index, test in enumerate(tests['input']):
        resp = is_friday_13(test[0], test[1])
        expected = tests['output'][index]
        if resp != expected:
            errors += 1
            print(f"\n\noriginal: {test}")
            print(resp)
            print(f"expected: {expected}")

    print(
        f"\nTests{' not ' if errors != 0 else ' '}passed, {errors} errors\n")


test_is_friday_13()
